src/progress.rs

```diff
@@ -6,14 +7,14 @@ use jujutsu_lib::git;
 use crate::ui::Ui;
 
 pub struct Progress<'a> {
-    ui: &'a mut Ui,
+    ui: &'a Mutex<&'a mut Ui>,
```

Comment:
I think it might be cleaner to pass in a `&mut Ui` to `update()`. That way `Progress` doesn't need to know that the `Ui` may be stored in a mutex (and doesn't require that *all* users store their `Ui` instance in a mutex).

Reply:
That makes it trickier to erase the bar on drop, but maybe we could do that on reaching 100% instead? Will experiment.

Reply:
Works great!